File: library/src/main/java/com/github/sundeepk/compactcalendarview/Events.java
package com.github.sundeepk.compactcalendarview;

import com.github.sundeepk.compactcalendarview.domain.Event;

import java.util.List;

class Events {

    private final List<Event> events;
    private final long timeInMillis;

    Events(long timeInMillis, List<Event> events) {
        this.timeInMillis = timeInMillis;
        this.events = events;
    }

    long getTimeInMillis() {
        return timeInMillis;
    }

    List<Event> getEvents() {
        return events;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Events event = (Events) o;

        if (timeInMillis != event.timeInMillis) return false;
        if (events != null ? !events.equals(event.events) : event.events != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = events != null ? events.hashCode() : 0;
        result = 31 * result + (int) (timeInMillis ^ (timeInMillis >>> 32));
        return result;
    }

    @Override
    public String toString() {
        return "Events{" +
                "events=" + events +
                ", timeInMillis=" + timeInMillis +
                '}';
    }
}

File: library/src/main/java/com/github/sundeepk/compactcalendarview/Events.kt
package com.github.sundeepk.compactcalendarview

import com.github.sundeepk.compactcalendarview.domain.Event

class Events(val timeInMillis: Long, val events: MutableList<Event>) {

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val event = o as Events
        if (timeInMillis != event.timeInMillis) return false
        return events == event.events
    }

    override fun hashCode(): Int {
        var result = events.hashCode() ?: 0
        result = 31 * result + (timeInMillis xor (timeInMillis ushr 32)).toInt()
        return result
    }

    override fun toString(): String {
        return "Events{" +
                "events=" + events +
                ", timeInMillis=" + timeInMillis +
                '}'
    }
}
